watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -191,9 +216,22 @@ def eq_flow_mass_comp(blk, j):
     dye_sep.s01 = Arc(
         source=dye_sep.P1.outlet, destination=dye_sep.nanofiltration.inlet
     )
-    dye_sep.s02 = Arc(
-        source=dye_sep.nanofiltration.byproduct, destination=m.fs.dye_retentate.inlet
-    )
+    if hasattr(m.fs, "dewater"):
+        dewater.s01 = Arc(
+            source=dye_sep.nanofiltration.byproduct, destination=dewater.dewaterer.inlet
+        )
+        # TODO: Recycle centrate stream back to the feed via a mixer
+        dewater.s02 = Arc(
```

Comment:
@adam-a-a I've tried adding this recycle loop and RO was failing to initialize even after playing around with tear guesses. So I figured I'd just leave this centrate as a product for now so that I can generate some numbers for tomorrow. One thing I did notice is that the RO permeate isn't being recycled either - is there a reason for this? If so, maybe it doesn't make sense to recycle this centrate stream either.

Reply:
@MarcusHolly Recycling the centrate stream back to the "headworks" of the plant is something that seems practical and closer to reality in my view. The RO permeate on the other hand would be recycled back to the textile plant for reuse, and this can just be represented by connecting the RO permeate stream with a Product block.

Reply:
We might have an issue similar to what @bknueven identified for on BSM2, where there could be some degeneracy introduced by a redundant constraint on the Mixer. Mind you, I haven't revisited the code yet to confirm, but it's worth mentioning. In any case, we can leave this as a to-do item for later.